plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/errors/VerificationError.kt

```diff
@@ -53,9 +55,32 @@ fun <I> VerificationError.getInfoOrNull(): I? =
     }
 
 /**
- * If the reason's offending node is a function application, then fetch the info metadata from the index-th argument.
+ * Extract the[Info] metadata from a [viper.silver.ast.MethodCall] argument.
  */
-fun ErrorReason.extractInfoFromFunctionArgument(argIndex: Int): Info = when (val node = reason.offendingNode()) {
-    is viper.silver.ast.FuncApp -> Info.fromSilver(node.args.apply(argIndex).info)
-    else -> error("The reason's offending node is not a function application.")
-}
+fun VerificationError.extractInfoFromFunctionArgument(argIndex: Int, isFromReason: Boolean = false): Info =
+    extraInfoFromCallableArgument<viper.silver.ast.FuncApp>(argIndex, isFromReason) { args() }
+
+/**
+ * Extract the [Info] metadata from the `argIndex`-th argument of a callable.
+ * @param argIndex The n-th callable argument from where extract the metadata.
+ * @param isFromReason If the info metadata is contained in the error's reason offending node.
+ * @param body Lambda function defining how to access the callable's argument list.
+ */
+private inline fun <reified T> VerificationError.extraInfoFromCallableArgument(
+    argIndex: Int,
+    isFromReason: Boolean,
```

Comment:
Is this really a convenient interface?  It seems like the structure you get is quite non-obvious; I think a wrapper around the error with properties like `reason` would be cleaner, so you can directly write something like `error.reason.asCallable().arg(index)`.

(Perhaps the restructuring of the error reporting and the addition of these functions would be better to handle separately?)

Reply:
I reverted these changes to their "original" state! I will create a new PR with new modification needed for #193 later.